### PR TITLE
Mapping featuress

### PR DESCRIPTION
**Pull request by Bod9001**
### Changelog:

CL: [New] added device attribute namer that allows you to rename items and set whether or not they will be saved in mapping mode
CL: [Improvement] copy and paste Will highlight what won't be saved
CL: [Improvement] VV hierarchy has been separated into separate window 
CL: [Improvement] VV you can now search variables
CL: [Balance] masks will protect you for one try before they are dropped on the floor from a face hugging attempt

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/Others/RandomItemSpot.cs [hunk 1]
```diff
@@ -107,7 +107,7 @@ private void GenerateItem(PoolData poolData, bool spawn)
 			}
 
 			var item = poolData.RandomItemPool.Pool.PickRandom();
-			var spread = fanOut ? Random.Range(-0.5f, 0.5f) : (float?) null;
+			var spread = fanOut ? Random.Range(0, 0.5f) : (float?) null;
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Items/Others/RandomItemSpot.cs:110]: why?

File: UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/DeviceAttributeEditor.cs [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+		if (Attribute != null)
+		{
+			IsMappedToggle.isOn = PressedObject.GetComponent<Attributes>().IsMapped;
+		}
+
+		SetNameUI();
+	}
+
+
+	public void OnMouseButtonUp()
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/DeviceAttributeEditor.cs]: codacy

File: UnityProject/Assets/Scripts/Items/Others/RandomItemSpot.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Others/RandomItemSpot.cs:110]: -is reserved for custom behaviour

**Comment by corp-0** [comment]
What do you mean with the last change in your change log?

**Comment by Bod9001** [comment]
basically when the face hugger first jumps you lose all your head equipment if you have any, 
however you don't get instantly hugged, 
however if there's another one ready to jump your screwed

File: UnityProject/Assets/Scripts/Items/Others/RandomItemSpot.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Others/RandomItemSpot.cs:110]: and due to how the code works 0.5f makes it a random direction in any direction

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Others/RandomItemSpot.cs:110]: since it goes random(spread, -spread )

File: UnityProject/Assets/Scripts/NPC/AI/Hostile/FaceHugAction.cs [hunk 3]
```diff
@@ -115,6 +115,7 @@ public static bool HasAntihuggerItem(Equipment equipment)
 					if (!Validations.HasItemTrait(item.gameObject, CommonTraits.Instance.AntiFacehugger))
 					{
 						Inventory.ServerDrop(itemSlot);
+						antiHugger = true; //It just tries again
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/NPC/AI/Hostile/FaceHugAction.cs]: Why is this change needed?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/NPC/AI/Hostile/FaceHugAction.cs]: I would say it's a balancing thing so it's relatively subjective,
 I think part of the issue is that we have such low pop that  the dancing idea of "well the station has loads of crew they can just bum rush the aliens doesn't really work" so this gives you an opportunity if you're skilled you can somewhat survive by yourself
 
 but it subjective so I'm interested to hear your opinion

Comment (corp-0) [reply UnityProject/Assets/Scripts/NPC/AI/Hostile/FaceHugAction.cs]: I think our mobs are already really dumb and easily cheesed. If they don't offer a challenge or aren't a threat then they might as well not be there.

This is specially important in very low pop scenarios, because the players need an antagonist and these games normally turn to PvE.

Comment (corp-0) [reply UnityProject/Assets/Scripts/NPC/AI/Hostile/FaceHugAction.cs]: Do you remember old xenos? They would chase you down with such an aggression that players actually felt threatened by them. They would break windows and tables if they had eyes on you.

Nowadays people can grab the xeno queen and throw her to disposals like its nothing.

Comment (Bod9001) [reply UnityProject/Assets/Scripts/NPC/AI/Hostile/FaceHugAction.cs]: yeah,  though that's mainly because we stuck between two systems currently

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/NPC/AI/Hostile/FaceHugAction.cs]: quick interjection
don't fix or work on old mobs, they're deprecated.
nuke them
work on facehuggers and xenos from scratch with a better base that's not slop
#10500

File: UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/DevCameraControls.cs [hunk 4]
```diff
@@ -300,10 +320,61 @@ public void OnEscape()
...
 	}
 
+	void ToggleMatrixCheck(bool state)
+	{
+		MatrixCheckerState = state;
+		if (state)
+		{
+			foreach (var Matrix in	MatrixManager.Instance.ActiveMatrices)
+			{
+				var colour =MatrixColours.PickRandom(); ;
```
Comment (wraith-54321) [inline UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/DevCameraControls.cs]:
```suggestion
				var colour =MatrixColours.PickRandom();
```

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Data/GasMix.cs [hunk 5]
```diff
@@ -34,7 +33,7 @@ public float Pressure
 			get => pressure;
 			set
 			{
-				if (float.IsNormal(value) == false && value != 0)
+				if (value.IsUnreasonableNumber() && value != 0)
```
Comment (wraith-54321) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Data/GasMix.cs]:
```suggestion
				if (value.IsUnreasonableNumber() && value != 0f)
```
